### DISCUSSION TITLE
Noticable delay when opening a dropdown with a larger number of items (50+)

### DISCUSSION THREAD

**Discussion by danvod** [taiga-family/taiga-ui#627] [Q&A] [answered]
I noticed the following when working with ComboBox and Select.

Basically, when there is a small number of items, everything works snappy and fluid - dropdown opens as soon as it is clicked.

However, on a larger number of items, say 50+, there is a noticable delay until dropdown appears.

For example, notice how under MultiSelect string array [example ](https://taiga-ui.dev/components/multi-select#string-array) everything is snappy and fluid.

However, if in stackblitz for that example you just duplicate items until you have 50+ in the array [(like so),](https://stackblitz.com/edit/angular-czuz3m-drjdt6) notice a delay until dropdown is opened - it is not huge, but it is noticable. I am aware that dropdowns generally should not have many items, but 50 items does not seem like _that_ much.

Further increasing the number of items makes the delay more noticable.

Any ideas / tips on how to solve this?

**Comment by waterplea** [comment, marked as answer]
Angular Dev mode is much slower than Prod mode due to debug overhead and double change detection cycles. Are you seeing this on Dev mode? Not sure there's much to be optimized at the moment, everything seems to be required for the flexibility options list has at the moment... 🤔

**Reply by danvod** [reply]
That makes sense, I'll try the same setup in both dev and prod mode and inspect the behavior. Thanks for replying! 😄
